vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/ConsolidateDescandantChild.java

```diff
@@ -0,0 +1,122 @@
+
+    @Override
+    public boolean rewritePre(Mutable<ILogicalOperator> opRef, IOptimizationContext context) throws AlgebricksException {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    public boolean rewritePost(Mutable<ILogicalOperator> opRef, IOptimizationContext context)
+            throws AlgebricksException {
+
+        boolean modified = false;
+        ILogicalOperator lo = opRef.getValue();
+        if (!(lo.getOperatorTag().equals(LogicalOperatorTag.UNNEST))) {
+            return modified;
+        }
+        AbstractUnnestOperator auo = (AbstractUnnestOperator) lo;
+        Mutable<ILogicalExpression> childExpression = ExpressionToolbox.findFirstFunctionExpression(
+                auo.getExpressionRef(), BuiltinOperators.CHILD.getFunctionIdentifier());
+        if (childExpression == null) {
+            return modified;
+        }
+        AbstractFunctionCallExpression childFnCall = (AbstractFunctionCallExpression) childExpression.getValue();
+        List<Mutable<ILogicalExpression>> list = (List<Mutable<ILogicalExpression>>) childFnCall.getArguments();
+        Mutable<ILogicalExpression> mle = (Mutable<ILogicalExpression>) (list).get(0);
+        ILogicalExpression le = mle.getValue();
+        if (!(le.getExpressionTag().equals(LogicalExpressionTag.VARIABLE))) {
+            return modified;
+        }
+        VariableReferenceExpression varLogicalExpression = (VariableReferenceExpression) le;
+        Mutable<ILogicalOperator> lop = OperatorToolbox.findProducerOf(opRef,
+                varLogicalExpression.getVariableReference());
+        ILogicalOperator lop1 = lop.getValue();
+        if (!(lop1.getOperatorTag().equals(LogicalOperatorTag.UNNEST))) {
+            return modified;
+        }
+        if (OperatorToolbox.getExpressionOf(lop, varLogicalExpression.getVariableReference()) == null) {
+            return modified;
+        }
+        ILogicalExpression variableLogicalExpression = (ILogicalExpression) OperatorToolbox.getExpressionOf(lop,
+                varLogicalExpression.getVariableReference()).getValue();
+        if (!(variableLogicalExpression.getExpressionTag().equals(LogicalExpressionTag.FUNCTION_CALL))) {
+            return modified;
+        }
+        AbstractFunctionCallExpression afce = (AbstractFunctionCallExpression) variableLogicalExpression;
+        if (!(afce.getFunctionIdentifier().equals(BuiltinOperators.DESCENDANT_OR_SELF.getFunctionIdentifier()))) {
+            return modified;
+        }
+        if (changeChildToDesc(opRef, childFnCall, afce, context)) {
+            modified = true;
+        }
+        return modified;
+    }
+
+    private boolean changeChildToDesc(Mutable<ILogicalOperator> opRef, AbstractFunctionCallExpression child,
+            AbstractFunctionCallExpression desc, IOptimizationContext context) {
+
+        child.setFunctionInfo(BuiltinOperators.DESCENDANT_OR_SELF);
```

Comment:
After inlining this function add a comment about the following statements change the plan. (All conditions have been meet.)